modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/HTTPResponseCodeTestCase.java

```diff
@@ -18,58 +18,92 @@
-import java.net.UnknownHostException;
-import java.util.HashMap;
-import java.util.Map;
-
-import static org.testng.Assert.assertEquals;
+import org.wso2.esb.integration.common.utils.ESBTestCaseUtils;
 
 public class HTTPResponseCodeTestCase extends ESBIntegrationTest {
     private Log log = LogFactory.getLog(HTTPResponseCodeTestCase.class);
+	private HttpServer server = null;
 
     @BeforeClass(alwaysRun = true)
     public void init() throws Exception {
         super.init();
-        loadESBConfigurationFromClasspath("/artifacts/ESB/synapseconfig/esbjava2283/synapse.xml");
+	    String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
+	    ESBTestCaseUtils util = new ESBTestCaseUtils();
+	    relativePath = relativePath.replaceAll("[\\\\/]", File.separator);
+	    OMElement apiConfig = util.loadResource(relativePath);
+	    addApi(apiConfig);
     }
 
-    @Test(groups = {"wso2.esb"}, description = "Test different response codes",
-          dataProvider = "getResponseCodes")
-    public void testReturnResponseCode(int responseCode) throws Exception {
-        int port = 9005;
-        String contentType = "text/xml";
-        SimpleWebServer simpleWebServer = new SimpleWebServer(port, responseCode);
-        log.info("Running the test for context type -" + contentType);
-        try {
-            simpleWebServer.start();
-            SimpleHttpClient httpClient = new SimpleHttpClient();
-            Map<String, String> headers = new HashMap<String, String>();
-            headers.put("Content-Type", contentType);
-            HttpResponse response = httpClient.doGet(context.getContextUrls().getServiceUrl() + "/passThoughProxy", headers);
-            log.info(response.getEntity().getContentType());
-            log.info(response.getStatusLine().getStatusCode());
-            simpleWebServer.terminate();
+	@Test(groups = { "wso2.esb" }, description = "Test different response codes", dataProvider = "getResponseCodes")
+	public void testReturnResponseCode(int responseCode) throws Exception {
+		int port = 8089;
+		server = HttpServer.create(new InetSocketAddress(port), 0);
+		server.createContext("/gettest", new MyHandler());
+		server.setExecutor(null); // creates a default executor
+		server.start();
+		switch (responseCode) {
+			case 200:
+				String contentType = "text/xml";
+				String url = "http://localhost:8280/serviceTest/test";
+				sendRequest(url, 200, contentType);
+			case 404:
+				contentType = "text/html";
+				url = "http://localhost:8280/serviceTest/notfound";
+				sendRequest(url, 404, contentType);
+		}
 
-            assertEquals(response.getEntity().getContentType().getValue(), contentType, "Expected content type doesn't match");
-            assertEquals(response.getStatusLine().getStatusCode(), responseCode, "response code doesn't match");
-        } finally {
-            simpleWebServer.terminate();
-            waitForPortCloser(port);
-        }
-    }
+		server.stop(0);
```

Comment:
Isn't it better to add this in finally block? So that we can make sure it stops regardless of what happens.